analyzer/src/main/kotlin/managers/Sbt.kt

```diff
@@ -213,3 +213,22 @@ class Sbt(
         // This is not implemented in favor over overriding [resolveDependencies].
         throw NotImplementedError()
 }
+
+private fun moveGeneratedPom(pomFile: File): File {
+    val targetDir = pomFile.absoluteFile.findClosestAncestor("target")?.parentFile ?: return pomFile
+    val targetFilename = pomFile.relativeTo(targetDir).invariantSeparatorsPath.replace('/', '-')
+    val targetFile = targetDir.resolve(targetFilename)
+
+    if (!pomFile.renameTo(targetFile)) {
+        Sbt.log.error { "Moving '${pomFile.absolutePath}' to '${targetFile.absolutePath}' failed." }
+        return pomFile
+    }
+
+    return targetFile
+}
+
+private fun File.findClosestAncestor(filename: String): File? {
```

Comment:
Can we instead make use of the existing `File.searchUpwardsForSubdirectory()` extension function here?